qcodes/dataset/data_set.py

```diff
@@ -708,21 +734,45 @@ def get_data(self,
             be of the datatypes stored in the database (numeric, array or
             string)
         """
-        valid_param_names = []
-        for maybeParam in params:
-            if isinstance(maybeParam, str):
-                valid_param_names.append(maybeParam)
-                continue
-            else:
-                try:
-                    maybeParam = maybeParam.name
-                except Exception as e:
-                    raise ValueError(
-                        "This parameter does not have  a name") from e
-            valid_param_names.append(maybeParam)
-        data = get_data(self.conn, self.table_name, valid_param_names,
+        valid_param_names = self._validate_parameters(*params)
+        if len(valid_param_names) == 0:
```

Comment:
This check does not catch the error in `get_parameter_data`, would it not be better to move this into the `sqlite_base` `get_data` ?

Reply:
ok now I see, sorry, yes it is indeed better to have it in `sqlite_base`.

Reply:
I will move it and let the function return an empty list plus issue a warning.